x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IlmHealthIndicatorService.java

```diff
@@ -53,11 +79,103 @@ public class IlmHealthIndicatorService implements HealthIndicatorService {
+    public static final List<HealthIndicatorImpact> AUTOMATION_DISABLED_IMPACT = List.of(
+        new HealthIndicatorImpact(
+            NAME,
+            AUTOMATION_DISABLED_IMPACT_ID,
+            3,
+            "Automatic index lifecycle and data retention management is disabled. The performance and stability of the cluster "
+                + "could be impacted.",
+            List.of(ImpactArea.DEPLOYMENT_MANAGEMENT)
+        )
+    );
+
+    public static final List<HealthIndicatorImpact> STAGNATING_INDEX_IMPACT = List.of(
+        new HealthIndicatorImpact(
+            NAME,
+            STAGNATING_INDEX_IMPACT_ID,
+            3,
+            "Automatic index lifecycle and data retention management cannot make progress on one or more indices. The performance and "
+                + "stability of the indices and/or the cluster could be impacted.",
+            List.of(ImpactArea.DEPLOYMENT_MANAGEMENT)
+        )
+    );
+
+    private static final TimeValue ONE_DAY = TimeValue.timeValueDays(1);
+
+    static final Map<String, RuleConfig> RULES_BY_ACTION_CONFIG = Map.of(
+        RolloverAction.NAME,
+        actionRule(RolloverAction.NAME).stepRules(
+            stepRule(WaitForActiveShardsStep.NAME, ONE_DAY),
+            stepRule(WaitForRolloverReadyStep.NAME, ONE_DAY),
+            stepRule(RolloverStep.NAME, ONE_DAY)
+        ),
+        //
+        MigrateAction.NAME,
+        actionRule(MigrateAction.NAME).maxTimeOnAction(ONE_DAY).noStepRules(),
+        //
+        SearchableSnapshotAction.NAME,
+        actionRule(SearchableSnapshotAction.NAME).maxTimeOnAction(ONE_DAY)
+            .stepRules(
+                stepRule(WaitForDataTierStep.NAME, ONE_DAY),
+                stepRule(WaitForIndexColorStep.NAME, ONE_DAY),
+                // The no-follower step is added here because an `UnfollowAction` is added before the `shrinkAction` in the follower cluster
+                stepRule(WaitForNoFollowersStep.NAME, ONE_DAY)
+            ),
+        //
+        DeleteAction.NAME,
+        actionRule(DeleteAction.NAME).stepRules(stepRule(DeleteStep.NAME, ONE_DAY)),
+        //
+        ShrinkAction.NAME,
+        actionRule(ShrinkAction.NAME).maxTimeOnAction(ONE_DAY)
+            .stepRules(
+                // The no-follower step is added here because an `unfollowAction` is added before the `shrinkAction` in the follower
+                // cluster.
+                stepRule(WaitForNoFollowersStep.NAME, ONE_DAY)
+            ),
+        //
+        AllocateAction.NAME,
+        actionRule(AllocateAction.NAME).maxTimeOnAction(ONE_DAY).noStepRules(),
+        //
+        ForceMergeAction.NAME,
+        actionRule(ForceMergeAction.NAME).maxTimeOnAction(ONE_DAY)
+            .stepRules(
+                stepRule(WaitForIndexColorStep.NAME, ONE_DAY),
+                stepRule(ForceMergeStep.NAME, ONE_DAY),
+                stepRule(SegmentCountStep.NAME, ONE_DAY)
```

Comment:
Same as above.

I believe we should tack number of retries for these ones (as they're async steps that move into the step from ERROR when they're retried)

Please update them to **also** track max 100 retries

Reply:
tackled on #96855